lib/codecs/UntypedLiteralEncoder.js

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) 2024 Digital Bazaar, Inc. All rights reserved.
+ */
+import {Token, Type} from 'cborg';
+import {CborldEncoder} from './CborldEncoder.js';
+import {CborldError} from '../CborldError.js';
+
+const TYPE_TO_TOKEN = new Map([
+  ['string', 0],
+  ['number', 1],
+  ['boolean', 2]
+]);
+
+export class UntypedLiteralEncoder extends CborldEncoder {
+  constructor({value, compressionMap} = {}) {
+    super();
+    this.value = value;
+    this.compressionMap = compressionMap;
+  }
+
+  encode() {
+    const {value, compressionMap} = this;
+    const type = typeof value;
+    const tokenValue = TYPE_TO_TOKEN.get(type);
+    if(tokenValue === undefined) {
+      throw new CborldError(
+        'ERR_UNSUPPORTED_JSON_TYPE',
+        `Unsupported value type "${type}".`);
+    }
+    if(type === 'string') {
+      if(compressionMap.has(value)) {
+        const compressedValue = compressionMap.get(value);
+        // FIXME: size of bytes?
+        const buffer = new ArrayBuffer(4);
```

Comment:
We're going to want to figure this out before test vectors happen. Do we want to do this in stages? So we support Uint8, Uint16, Uint32? We should throw if we're larger than that right now so we don't have some weird problem. We could extend support for larger integers in the future.